### DISCUSSION TITLE
How to ignore fields like dates and IDs?

### DISCUSSION THREAD

**Discussion by alvinwo** [origin-energy/java-snapshot-testing#88] [Q&A] [answered]
Hey team, this is a great project to do the snapshot testing in Java world. I noticed in the doc that you support to ignore fields like dates but I didn't find examples about this.

So if I would like to ignore some fields in snapshot testing, should I create a new SnapshotSerializer, or it there any other way to achieve that?

**Comment by jackmatt2** [comment, marked as answer]
Yes, you can use regular Jackson configuration to ignore dates etc.

You will need to create a serializer similar to this Hibernate one.
https://github.com/origin-energy/java-snapshot-testing#example-hibernateserializer

Then customise as needed
```java
objectMapper.addMixIn(DateTime.class, IgnoreTypeMixin.class);
objectMapper.addMixIn(Instant.class, IgnoreTypeMixin.class);
objectMapper.addMixIn(Date.class, IgnoreTypeMixin.class);
```

Update your `snapshot.properties` to include the new serializer (alternatively replace the default one)
```
serializer.custom=com.example.MyCustomSerializer
```

```
expect.serializer('custom').toMatchSnapshot(foo)
```

**Reply by alvinwo** [reply]
Got it, thx
